src/chain/chaininterface.rs

```diff
@@ -11,14 +11,12 @@ use std::sync::atomic::{AtomicUsize, Ordering};
 
 /// Used to give chain error details upstream
 pub enum ChainError {
-	/// Chain isn't supported
+	/// Client doesn't support UTXO lookup (but the chain hash matches our genesis block hash)
```

Comment:
Hmm, I think we can have cases where the chain hash is really unknown and don't match our genesis block hash. Or maybe we want to filter channel_MSG to router by chain downstream ?

Reply:
I'm not entirely sure what you're suggesting here? If a client is eg an SPV client they may not want to support arbitrary UTXO lookup, but we still want to check the genesis block hash is correct before we handle the message, hence the distinction here.

Reply:
Ok got it!